agent/proxycfg/state.go

```diff
@@ -303,6 +307,17 @@ func (s *state) run(ctx context.Context, snap *ConfigSnapshot) {
 	sendCh := make(chan struct{})
 	var coalesceTimer *time.Timer
 
+	scheduleUpdate := func() {
+		r := s.rateLimiter.Reserve()
+
+		coalesceTimer = time.AfterFunc(coalesceTimeout+r.Delay(), func() {
```

Comment:
Should this instead wait for `max(coalesceTimeout, r.Delay())`?

coalesce timeout will ensure this states snapshot doesn't get emitted too frequently
r.Delay ensures that we don't emit snapshots too frequently globally.

This might be sort of a premature optimization but figured I would note it anyways.

Reply:
Good point! I think either is fine honestly, but I guess using the `max` could save 200ms of unnecessary waiting - will change it 🙇🏻‍♂️